icepyx/core/temporal.py

```diff
@@ -0,0 +1,490 @@
+import datetime as dt
+import warnings
+
+
+'''
+Helper functions for validation of dates
+'''
+
+
+def convert_string_to_date(date):
+    """
+    Converts a string to a datetime object.
+    Throws an error if an invalid format is passed in.
+
+    Parameters
+    ----------
+    date: string
+        A string representation for the date value. Current supported date formats are:
+            * "YYYY-MM-DD"
+            * "YYYY-DOY"
+    Returns
+    -------
+    datetime object, representing the date from the string parameter.
+
```

Comment:
need to check docstrings for this new module:
- [ ] confirm ordering of entries
- [ ] add missing components (e.g. for properties)